### PR TITLE
Pass on the return value of file_put_content

### PR DESCRIPTION
**Pull request by rullzer**
This will help with AppData hardening etc

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

**Review by icewind1991 — CHANGES_REQUESTED** [review]
An exception should be thrown when it returns false instead

**Comment by rullzer** [comment]
@icewind1991 any suggestions for which exception to use?

**Comment by icewind1991** [comment]
You can only use something generic here since we dont know the failure cause, not sure what would be best though

**Comment by rullzer** [comment]
@icewind1991 ok added an exception.

I introduced the `GenericFileException` (fancy name I know). Maybe we should use this in case nothing specific matches? ALso we could base all othe file exceptions on it. Just so there is more of a hierarchy there.

File: lib/private/Files/Node/File.php [hunk 1]
```diff
@@ -57,11 +58,14 @@ public function getContent() {
 	/**
 	 * @param string $data
 	 * @throws \OCP\Files\NotPermittedException
+	 * @throws \OCP\Files\GenericFileException
 	 */
 	public function putContent($data) {
 		if ($this->checkPermissions(\OCP\Constants::PERMISSION_UPDATE)) {
 			$this->sendHooks(array('preWrite'));
-			$this->view->file_put_contents($this->path, $data);
+			if ($this->view->file_put_contents($this->path, $data)) {
```
Comment (icewind1991) [inline lib/private/Files/Node/File.php]: missing a `!` :see_no_evil:

**Review by icewind1991 — CHANGES_REQUESTED** [review]

File: lib/private/Files/Node/File.php (hunk 1 above)
Comment (rullzer) [reply lib/private/Files/Node/File.php]: fixed...

**Review by icewind1991 — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]

**Review by juliusknorr — APPROVED** [review]

**Pull request merged**